runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/GcsStager.java

```diff
@@ -67,4 +78,19 @@ public static GcsStager fromOptions(PipelineOptions options) {
         options.getStagingLocation(),
         createOptions);
   }
+
+  private void warnIfStagingHasTTL() {
+    try {
+      LOG.debug("Checking if staging location {} has TTL assigned", options.getStagingLocation());
+      boolean stagingHasTTL = options.as(GcsOptions.class).getGcsUtil()
+          .bucketHasTTL(GcsPath.fromUri(options.getStagingLocation()));
+      if (stagingHasTTL) {
+        LOG.warn("Staging location {} has TTL assigned. This might cause unpredictable bugs."
+            , options.getStagingLocation());
+      }
+    } catch (Exception ie) {
+      LOG.warn("Exception while trying to determine if staging location has TTL assigned", ie);
```

Comment:
`"Could not determine if staging location {} has an assigned TTL", options.getStagingLocation(), ie`